kala.py

```diff
@@ -37,43 +39,185 @@
     db=os.environ.get('KALA_MONGODB_DB', app.config['mongodb.db']),
     json_mongo=True))
 
-
 if os.environ.get('KALA_CORS_ENABLE', app.config['cors.enable']):
     @app.hook('after_request')
     def add_cors_response_headers():
-        if bottle.request.method in ('GET', 'OPTIONS'):
+        if bottle.request.method in ('GET', 'OPTIONS', 'POST'):
             bottle.response.set_header('Access-Control-Allow-Origin', '*')
             bottle.response.set_header('Access-Control-Allow-Headers', ','.join(CORS_HEADERS))
 
+if 'filter.json' in app.config:
+    with open(app.config['filter.json']) as data_file:
+        app.config['filter.json'] = json.load(data_file)
+if 'filter.read' in app.config:
+    app.config['filter.read'] = app.config['filter.read'].split(',') \
+        if app.config['filter.read'] else None
+
 
 def _get_json(name):
     result = bottle.request.query.get(name)
     return json.loads(result) if result else None
 
 
-@app.route('/<collection>')
+def _filter_write(mongodb, document):
+    # This will throw if setting isn't found in config.
+    # Expected as your filter won't work without filter JSON path.
+    filter_json = os.environ.get('KALA_FILTER_JSON', app.config['filter.json'])
```

Comment:
Yes, not sure what I was doing here. Was in a bit of a rush I think.
Filter_Json should be the json string.